derive/src/rpc_trait.rs

```diff
@@ -131,13 +132,15 @@ fn generate_rpc_item_trait(item_trait: &syn::ItemTrait) -> Result<(syn::ItemTrai
 
 	let to_delegate_method =
 		generate_trait_item_method(&method_registrations, &item_trait, rpc_trait.has_metadata, has_pubsub_methods)?;
+
+  let rpc_client_module = generate_client_module(&method_registrations, &item_trait)?;
```

Comment:
Can we make this optional, maybe a parameter to the main `#[rpc]` attribute? Otherwise it will generate unused code warnings, no?

Reply:
To support generics we also need information about the implementations, to figure out the concrete types. I think it's probably a good idea to create a json file of the required information for generating a client as an intermediate step. Then generating the client would be optional by using something like `generate_client_from_json!()`.

Reply:
Now that we are using a struct for the client handling generics is quite easy, ignore my previous comment :)

Reply:
Do you prefer opt-in or opt-out?